lte/gateway/deploy/arm_build.yml

```diff
@@ -0,0 +1,189 @@
+
+---
+# Copyright 2021 The Magma Authors.
+
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+- name: Phase 1 - Create the arm graviton instance
+  hosts: localhost
+  gather_facts: False
+  vars:
+    arm_instance_type: t4g.medium
+    arm_security_group: sg-068f38e992f0ffb63
+    arm_ami_instance: ami-00a0488e9d5582804
+    arm_region: us-west-1
+    keypair: gha_keypair
+    arm_vpc: subnet-2674e740
+    aws_secret_key: "overwrite_me"
+    aws_access_key: "overwrite_me"
+    ansible_ssh_private_key_file: "~/.ssh/gha_keypair.pem"
+    ansible_ssh_user: "ubuntu"
+    magma_branch: "master"
+    magma_tag: ""
+  tasks:
+    - name: Launch instance
+      amazon.aws.ec2_instance:
+        name: "arm-build-{{ magma_branch }}-{{ magma_tag }}"
+        aws_access_key: "{{ aws_access_key }}"
+        aws_secret_key: "{{ aws_secret_key }}"
+        key_name: "{{ keypair }}"
+        security_group: "{{ arm_security_group }}"
+        instance_type: "{{ arm_instance_type }}"
+        image_id: "{{ arm_ami_instance }}"
+        wait: true
+        region: "{{ arm_region }}"
+        vpc_subnet_id: "{{ arm_vpc }}"
+        volumes:
+          - device_name: /dev/sda1
+            ebs:
+              volume_size: 50
+              delete_on_termination: true
+        network:
+          assign_public_ip: yes
+        tags:
+          Environment: CI
+      register: ec2
+
+    - name: Sleep for 10 seconds if AWS didn't return the public ip
+      ansible.builtin.wait_for:
+        timeout: 10
+      delegate_to: localhost
+      when: ec2.instances[0].public_ip_address is undefined
+
+    - name: Fetch the public ip address
+      amazon.aws.ec2_instance:
+        name: "arm-build-{{ magma_branch }}-{{ magma_tag }}"
+        aws_access_key: "{{ aws_access_key }}"
+        aws_secret_key: "{{ aws_secret_key }}"
+        key_name: "{{ keypair }}"
+        security_group: "{{ arm_security_group }}"
+        instance_type: "{{ arm_instance_type }}"
+        image_id: "{{ arm_ami_instance }}"
+        wait: true
+        region: "{{ arm_region }}"
+        vpc_subnet_id: "{{ arm_vpc }}"
+        volumes:
+          - device_name: /dev/sda1
+            ebs:
+              volume_size: 50
+              delete_on_termination: true
+        network:
+          assign_public_ip: yes
+        tags:
+          Environment: CI
+      register: ec2
+      when: ec2.instances[0].public_ip_address is undefined
```

Comment:
https://docs.ansible.com/ansible/latest/user_guide/playbooks_variables.html#registering-variables

Register is just an ansible action that save the output of a task in a variable, I could have called it whatever I want. 
As I use a specific naming  (name: "`arm-build-{{ magma_branch }}-{{ magma_tag }}"`)it will always return the same instance, that's why I can hardcode the first element of the array